src/tmi/mod.rs

```diff
@@ -86,6 +86,46 @@ impl<'a, C: crate::HttpClient<'a>> TMIClient<'a, C> {
             .map_err(|e| RequestError::Utf8Error(req.body().clone(), e))?;
         serde_json::from_str(text).map_err(Into::into)
     }
+
+    /// Get either the broadcaster that a given channel is hosting,
+    /// or the list of channels hosting a given target broadcaster.
+    ///
+    /// # Notes
+    /// This endpoint requires `hosts={id}` XOR `target={id}` in the query
+    /// (providing both will result in an error, therefore this function takes
+    /// a `HostsRequest` enum).
+    pub async fn get_hosts(
+        &'a self,
+        include_logins: bool,
```

Comment:
Why is this not on the request itself?

Reply:
I debated with myself on this... The main purpose of `HostsRequest` is to handle the `xor`-ness of this endpoint, so it's clear that both `host` and `target` cannot be set at the same time. But I couldn't think of a better name for the enum.

Putting `include_logins` in both of the enum variants seems slightly redundant, but the other option would be to rename the enum something like `ChannelId` and make `HostsRequest` a struct containing both it and `include_logins`.

Which would you prefer?

Reply:
I'm leaning towards renaming it to `HostsRequestId` myself. Would like something shorter, but I think this clearly indicates its purpose, and seems to result in the cleanest code. Ultimately what this wants to be is an anonymous sum type, but those don't exist in Rust yet. 

Putting `include_logins` inside the variants not only duplicates fields, but also makes it more awkward to match on. Logically, `include_logins` and `host | target` are separate parameters of the request and can be evaluated in sequence. Making it a struct would mean we still need to deal with the enum naming issue anyway.